### ISSUE TITLE
False positive: Expression of type "IntegerField[Unknown, Unknown] | int" cannot be assigned to return type "int"

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#949]
When django model has method modifying specified attribute, i get this error.

**Here is example model to reproduce error**

```python
from django.db import models


class Order(models.Model):
    status = models.IntegerField()

    def get_status(self) -> int:
        return self.status  # error: Expression of type "IntegerField[Unknown, Unknown] | int" cannot be assigned to return type "int"

    def set_status(self, new_status: int) -> None:
        self.status = new_status
```

If i remove "set_status", error above disappears.

I'm using django-stubs 1.5.0 and pyright 1.1.63 (cli version).

**Comment by erictraut** [comment]
Thanks for the bug report.

There are actually two separate bugs here. The first is a bug in the django type stubs that ship with pylance. I have filed a [separate bug](https://github.com/microsoft/pylance-release/issues/249) on this issue.

The second bug is in the pyright type checker itself. It was not properly handling the case where a member access expression (such as `self.status`) referenced a member whose type was a union of multiple classes _and_ where some of those classes provided special accessor methods (e.g. `__get__`) and some did not. I have fixed this bug in the type checker, and the fix will be included in the next release of Pyright and Pylance.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.64, which I just published. It will also be included in the next release of Pylance.
